api/client/client.go

```diff
@@ -52,11 +52,11 @@ func (t TokenAuth) GetRequestMetadata(ctx context.Context, _ ...string) (map[str
 
 // RequireTransportSecurity specifies if the connection should be secure.
 func (t TokenAuth) RequireTransportSecurity() bool {
-	return t.secure
+	return t.Secure
 }
 
-// NewClient creates a client.
-func NewClient(target string, optsOverrides ...grpc.DialOption) (*Client, error) {
+// CreateClientConn creates a gRPC connection with sensible defaults and the provided overrides.
+func CreateClientConn(target string, optsOverrides ...grpc.DialOption) (*grpc.ClientConn, error) {
```

Comment:
@jsign I just added this as well so the logic around options doesn't have to be re-created elsewhere.